src/main/scala/org/zalando/spearheads/innkeeper/routes/PostRoutes.scala

```diff
@@ -0,0 +1,60 @@
+package org.zalando.spearheads.innkeeper.routes
+
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.http.scaladsl.server.Directives.{ reject, handleWith, as, entity, post }
+import akka.http.scaladsl.server.{ AuthorizationFailedRejection, Route }
+import com.google.inject.Inject
+import org.slf4j.LoggerFactory
+import org.zalando.spearheads.innkeeper.RouteDirectives.{ isStrictRoute, isRegexRoute }
+import org.zalando.spearheads.innkeeper.UnmarshallRejection
+import org.zalando.spearheads.innkeeper.api.{ RouteOut, RouteIn }
+import org.zalando.spearheads.innkeeper.metrics.RouteMetrics
+import org.zalando.spearheads.innkeeper.oauth.OAuthDirectives.{ team, hasOneOfTheScopes }
+import org.zalando.spearheads.innkeeper.oauth.{ AuthenticatedUser, Scopes }
+import org.zalando.spearheads.innkeeper.services.{ ServiceResult, RoutesService }
+import org.zalando.spearheads.innkeeper.services.team.TeamService
+import org.zalando.spearheads.innkeeper.api.JsonProtocols._
+import akka.http.scaladsl.server.RouteConcatenation._
+import scala.concurrent.{ Future, ExecutionContext }
+
+/**
+ * @author dpersa
+ */
+class PostRoutes @Inject() (implicit val executionContext: ExecutionContext,
+                            val teamService: TeamService,
+                            val routesService: RoutesService,
+                            val metrics: RouteMetrics,
+                            val scopes: Scopes) {
+
+  private val LOG = LoggerFactory.getLogger(this.getClass)
+
+  def apply(authenticatedUser: AuthenticatedUser, token: String): Route = {
+    post {
+      LOG.info("post /routes/")
+      entity(as[RouteIn]) { route =>
+        LOG.debug(s"route ${route}")
+        team(authenticatedUser, token)(teamService) { team =>
+          (isRegexRoute(route.route) & hasOneOfTheScopes(authenticatedUser)(scopes.WRITE_REGEX)) {
```

Comment:
greedy and? Shouldn't the `&` be `&&` ? Currently it always eval hasOneOfTheScopes even if isRegexRoute is already false.